File: 序号/383.赎金信.cpp
class Solution {
public:
    /** 判断a能否由b构成，每个字符用一次 */
    bool canConstruct(string a, string b) {
        if(a.length()>b.length()) return false;
        int cnt[26]{0};
        for(char c:b){
          cnt[c-'a']++;
        }
        for(char c:a){
          if(cnt[c-'a']){
            cnt[c-'a']--;
          }else{
            return false;
          }
        }
        return true;
    }
};

File: 序号/383. 赎金信.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        cnt=[0]*128
        for x in magazine:
            cnt[ord(x)]+=1
        for x in ransomNote:
            cnt[ord(x)]-=1
            if cnt[ord(x)]<0:
                return False
        return True